### DISCUSSION TITLE
Add `page` to `reportWebVitals` payload

### DISCUSSION THREAD

**Discussion by jackjdleslie** [vercel/next.js#32221] [Ideas]
### Describe the feature you'd like to request

When using Vercel Analytics with Next.js you get access to the `page` for a particular metric. It would be nice if you could also get access to `page` when using the custom `reportWebVitals` function.

### Describe the solution you'd like

For the Vercel Analytics performance relayer you get the page via `__NEXT_DATA__.page` so by adding `page: window.__NEXT_DATA__.page` to the `reportWebVitals` payload we can achieve the same effect.

### Describe alternatives you've considered

`__NEXT_DATA__.page` looks like the easiest way to do this, but perhaps there's another way of getting this data

**Comment by balazsorban44** [comment]
what's the difference between this and `window.location.href`? Any benefits?

**Comment by jackjdleslie** [comment]
> what's the difference between this and `window.location.href`? Any benefits?

It will show the value as the Next.js definition of a page, rather than just the URL, which is good for dynamic routes as it would show `/id/[value]` rather than `id/123` and `id/abc` so you can collect the metrics for the route rather than specific instances of the route

This is what’s used in Vercel Analytics as well
